plugins/UM3NetworkPrinting/src/Cloud/CloudOutputDevice.py

```diff
@@ -0,0 +1,483 @@
+
+    ##  We need to override _createEmptyRequest to work for the cloud.
+    def _createEmptyRequest(self, path: str, content_type: Optional[str] = "application/json") -> QNetworkRequest:
+        # noinspection PyArgumentList
+        url = QUrl(path)
+        request = QNetworkRequest(url)
+        request.setHeader(QNetworkRequest.ContentTypeHeader, content_type)
+        request.setHeader(QNetworkRequest.UserAgentHeader, self._user_agent)
+        
+        if not self._account.isLoggedIn:
+            # TODO: show message to user to sign in
+            self.setAuthenticationState(AuthState.NotAuthenticated)
+        else:
+            # TODO: not execute call at all when not signed in?
+            self.setAuthenticationState(AuthState.Authenticated)
+            request.setRawHeader(b"Authorization", "Bearer {}".format(self._account.accessToken).encode())
+
+        return request
+    
+    ##  Set all the interface elements and texts for this output device.
+    def _setInterfaceElements(self):
+        self.setPriority(2)  # make sure we end up below the local networking and above 'save to file'
+        self.setName(self._id)
+        self.setShortDescription(self.I18N_CATALOG.i18nc("@action:button", "Print via Cloud"))
+        self.setDescription(self.I18N_CATALOG.i18nc("@properties:tooltip", "Print via Cloud"))
+        self.setConnectionText(self.I18N_CATALOG.i18nc("@info:status", "Connected via Cloud"))
+    
+    ##  Called when Cura requests an output device to receive a (G-code) file.
+    def requestWrite(self, nodes: List[SceneNode], file_name: Optional[str] = None, limit_mime_types: bool = False,
+                     file_handler: Optional[FileHandler] = None, **kwargs: str) -> None:
+        
+        # Show an error message if we're already sending a job.
+        if self._sending_job:
+            self._onUploadError(self.I18N_CATALOG.i18nc(
+                "@info:status", "Sending new jobs (temporarily) blocked, still sending the previous print job."))
+            return
+        
+        # Indicate we have started sending a job.
+        self._sending_job = True
+        self.writeStarted.emit(self)
+
+        file_format = self._determineFileFormat(file_handler)
+        writer = self._determineWriter(file_handler, file_format)
+        if not writer:
+            Logger.log("e", "Missing file or mesh writer!")
+            self._onUploadError(self.I18N_CATALOG.i18nc("@info:status", "Could not export print job."))
+            return
+
+        stream = io.StringIO() if file_format["mode"] == FileWriter.OutputMode.TextMode else io.BytesIO()
+        writer.write(stream, nodes)
+        self._sendPrintJob(file_name + "." + file_format["extension"], file_format["mime_type"], stream)
+
+    # TODO: This is yanked right out of ClusterUM3OutputDevice, great candidate for a utility or base class
+    def _determineFileFormat(self, file_handler) -> Optional[Dict[str, Union[str, int]]]:
```

Comment:
This entire function should be removed. The cloud should always support UFP. The only reason this is in the cloud plugin is because there are older firmware versions that don't support this.